test/NVQPP/bug_qubit.cpp

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// This code is from Issue 251.
+
+// RUN: nvq++ --enable-mlir -v %s --target quantinuum --quantinuum-url http://localhost:6245 && ( ./a.out > %t 2>&1 || ( cat %t | FileCheck %s ) )
```

Comment:
I think you can reproduce this without relying on the failure caused by no quantinuum mock server - just use `--target quantinuum --emulate`

Reply:
OK. I xeroxed the test case from #251 here and wasn't sure what, if anything, I should change.